### PR TITLE
dropout #23

### PR DESCRIPTION
**Pull request by ali-aboelezz**

### CODE REVIEW

**Comment by ali-aboelezz** [comment]
Close #23

**Comment by ali-aboelezz** [comment]
Close #10320

File: ivy_tests/test_ivy/test_frontends/test_torch/test_torch_nn_functional_dropout.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        min_dim_size=2,
+    ),
+    prob=helpers.floats(min_value=0, max_value=0.9),
+    training=st.booleans(),
+    test_with_out=st.just(False),
+    test_inplace=st.booleans(),
+)
+def test_torch_dropout(
+    *,
+    dtype_and_input,
```
Comment (zaeemansari70) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_torch_nn_functional_dropout.py:30]: this should be `dtype_and_x` as you are calculating `dtype_and_x` above in the decorator
your function is expecting `dtype_and_x` here.

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
Hey @ali-aboelezz ! 
Thank you so much for contributing to ivy! 🙂
Your PR looks amazing ✨!
Just requested a few changes and left comments on them which needs to be addressed before we can merge your PR! 🙂

Do let me know if you have any other questions!
Thanks!

File: ivy_tests/test_ivy/test_frontends/test_torch/test_torch_nn_functional_dropout.py (hunk 1 above)
Comment (ali-aboelezz) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_torch_nn_functional_dropout.py]: i made the changes you asked me for  please check the second commit @zaeemansari70

File: ivy/functional/frontends/torch/nn/functional/torch_nn_functional_dropout.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import ivy
+from ivy.func_wrapper import with_unsupported_dtypes
+
+from ivy.functional.frontends.torch.func_wrapper import to_ivy_arrays_and_back
+
+
+@to_ivy_arrays_and_back
+@with_unsupported_dtypes({"1.11.0 and below": ("float16",)}, "torch")
+
+def dropout(input_, p=0.5, training=True, inplace=False):
```
Comment (sherry30) [inline ivy/functional/frontends/torch/nn/functional/torch_nn_functional_dropout.py:11]: the argument is `input` not `input_`.
You can check the torch docs [here](https://pytorch.org/docs/stable/generated/torch.nn.functional.dropout.html)

**Review by sherry30 — CHANGES_REQUESTED** [review]
Hi,

Your PR looks good overall. A few changes are required:
1) There are a bunch of lint formatting errors, can you fix those.
2) requested the change on specific line.
3) the test is failing. This might get fixed if you fix the 2nd change requested since the function expects `input`. But you will have to look into any other failures if they occur.

Let me know if you need any help with anything,
Thanks

File: ivy/functional/frontends/torch/nn/functional/torch_nn_functional_dropout.py (hunk 2 above)
Comment (ali-aboelezz) [reply ivy/functional/frontends/torch/nn/functional/torch_nn_functional_dropout.py]: please check that 77cd844b43ab0e5d67bf6e4dc1e5d1b4128e3e73

**Review by sherry30 — CHANGES_REQUESTED** [review]
Hi,

The test function is passing now but there are some lint formatting errors.
I'll merge your PR as soon as you fix those 😄 

Thanks

File: ivy/functional/frontends/torch/nn/functional/torch_nn_functional_dropout.py (hunk 2 above)
Comment (ali-aboelezz) [reply ivy/functional/frontends/torch/nn/functional/torch_nn_functional_dropout.py]: lint formatting error , you mean error in the dropout function not the testing function ,right? @sherry30
